code/modules/projectiles/guns/energy.dm

```diff
@@ -128,3 +131,30 @@ var/global/list/registered_cyborg_weapons = list()
 			icon_state = "[modifystate][get_charge_ratio()]"
 		else
 			icon_state = "[initial(icon_state)][get_charge_ratio()]"
+
+
+//For removable cells.
+/obj/item/gun/energy/attack_hand(mob/user)
+	if(!user.is_holding_offhand(src)|| isnull(cell_type) || isnull(power_supply) )
+		return ..()
+	user.put_in_hands(power_supply)
+	power_supply = null
+	user.visible_message(SPAN_NOTICE("\The [user] unloads \his [src]."))
+	playsound(src,'sound/weapons/guns/interaction/smg_magout.ogg' , 50)
+	update_icon()
+
+/obj/item/gun/energy/attackby(var/obj/item/A, mob/user)
+	if(isnull(cell_type))
+		return
+	if(!user.is_holding_offhand(src))
+		to_chat(user, SPAN_NOTICE("Hold \The [src] in your hands to load it."))
+		return
+	if(istype(power_supply) )
+		to_chat(user, SPAN_NOTICE("There is already \a [power_supply] loaded."))
+
+	if(istype(A, cell_type) && do_after(user, 5, A, can_move = TRUE) && user.unEquip(A, src) )
+		A.forceMove(src)
+		power_supply = A
+		user.visible_message(SPAN_WARNING("\The [user] loads \his [src]!"))
```

Comment:
Needs to either be something like `\The [user] loads \the [A] into \the [src].` or you need to retrieve/use the pronoun decl from the user. `\his` is verboten.

Reply:
Addressed.